uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.php

```diff
@@ -142,8 +142,12 @@ public function postProcess() {
   private function getAbsencePeriodFromRequest() {
     $periodId = CRM_Utils_Request::retrieve('id', 'Integer');
     if(!$periodId) {
-      return AbsencePeriod::getCurrentPeriod();
+      $today = new DateTime('today');
+      $absencePeriods = AbsencePeriod::getPeriodsBetweenDates($today);
```

Comment:
From the name I would have expected `getCurrentPeriod` to return the same as this, and `getPeriodsBetweenDates` doesn't make much sense reading it when only one date is passed in. Not really relevant to this PR, but anyway.

Reply:
I think its fine to leave the method name as is, The method gets periods between two dates and when the second parameter is not there, it simply returns periods between the first date and the future.  I remember I had some conversation with Davi about the name too before settling for this when I added the method initially.

Reply:
After the latest changes this PR is pretty small and the code all looks fine for me. I'm not sure if I know enough about the purpose of absence periods, because the whole PR is a bit confusing to me. What are absence periods for?

I can see the point of avoiding a 500, but I'm not sure why leave requests should be shoehorned into an absence period that doesn't actually contain them.  The logic @davialexandre mentioned above sounds like it's getting more and more complicated, and complex business logic usually means we'll have confused users in a few months.

Reply:
@mickadoo, the issue is actually about job contracts and absence periods not leave requests and absence periods.
Entitlements are calculated for absence periods and if when redirecting to the user to the entitlements calculation page after creating/updating a contract, there is no absence period containing the current date, we will redirect user to calculate entitlements for the closest absence period available, There is a previous/next button that allows the user to navigate absence periods and calculate entitlements for which period they want to calculate for.
You can take a look at the original ticket that documents the bug https://compucorp.atlassian.net/browse/PCHR-2952.